mlflow/pyfunc/__init__.py

```diff
@@ -307,6 +307,7 @@
     _parse_requirements,
     warn_dependency_requirement_mismatches,
 )
+from pyspark.sql import DataFrame as SparkDataFrame
```

Comment:
Same comment as https://github.com/mlflow/mlflow/pull/10939/files#r1471882457. We can't import PySpark at the top level here because many developers use MLflow Models / pyfunc without PySpark